feature/policy_forwarding/otg_tests/mpls_gre_udp_decap_test/mpls_gre_udp_decap_test.go

```diff
@@ -0,0 +1,582 @@
+					outerIPSrc:  outerIpv6Src,
+					outerIPDest: outerIpv6DstB,
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if err := runTest(t, dut, top, tc, ate); err != nil {
+				t.Errorf("Test %s failed: %v", tc.name, err)
+			}
+		})
+	}
+}
+
+func runTest(t *testing.T, dut *ondatra.DUTDevice, config gosnappi.Config, tc testCase, ate *ondatra.ATEDevice) error {
+	t.Log("Configuring input policy")
+	configureInputPolicy(t, dut, tc)
+	var flowErrors []error
+	for _, flowConfig := range tc.flowConfigs {
+		if err := sendAndValidateTraffic(t, ate, config, tc, flowConfig); err != nil {
+			flowErrors = append(flowErrors, err)
+		}
+	}
+	if len(flowErrors) > 0 {
+		return errors.Join(flowErrors...)
+	}
+	return nil
+}
+
+func sendAndValidateTraffic(t *testing.T, ate *ondatra.ATEDevice, config gosnappi.Config, tc testCase, flowConfig flowConfig) error {
+	otg := ate.OTG()
+	var captureState gosnappi.ControlState
+	flowName := fmt.Sprintf("flow-%s-%s-%s", flowConfig.outerIPType, tc.encapType, flowConfig.innerIPType)
+	configureFlow(t, &config, tc, flowConfig, flowName)
+	enableCapture(t, ate, config, []string{capturePort})
+	defer clearCapture(t, ate, config)
+
+	otg.PushConfig(t, config)
+	otg.StartProtocols(t)
+	captureState = startCapture(t, ate)
+	otg.StartTraffic(t)
+	err := waitForTraffic(t, otg, flowName, trafficTimeout)
+	stopCapture(t, ate, captureState)
+	if err != nil {
+		return err
+	}
+	otg.StopProtocols(t)
+	otgutils.LogFlowMetrics(t, otg, config)
+	otgutils.LogPortMetrics(t, otg, config)
+
+	flowMetrics := gnmi.Get(t, ate.OTG(), gnmi.OTG().Flow(flowName).State())
+	if *flowMetrics.Counters.OutPkts == 0 {
+		return errors.New("no packets transmitted")
+
+	}
+
+	if *flowMetrics.Counters.InPkts != noOfPackets {
+		return fmt.Errorf("unexpected number of packets received: got %d, want %d", *flowMetrics.Counters.InPkts, noOfPackets)
+	}
+
+	processCapture(t, ate, capturePort, flowName)
+	if err := verifyReceivedInnerPacket(t, captureFilePath, tc, flowConfig); err != nil {
+		return fmt.Errorf("packet validation failed: %w", err)
+	}
+	return nil
+}
+
+func enableCapture(t *testing.T, ate *ondatra.ATEDevice, topo gosnappi.Config, otgPortNames []string) {
+	t.Helper()
+	for _, port := range otgPortNames {
+		t.Log("Enabling capture on ", port)
+		topo.Captures().Add().SetName(port).SetPortNames([]string{port}).SetFormat(gosnappi.CaptureFormat.PCAP)
+	}
+	ate.OTG().PushConfig(t, topo)
+}
+
+func clearCapture(t *testing.T, ate *ondatra.ATEDevice, topo gosnappi.Config) {
+	t.Helper()
+	topo.Captures().Clear()
+	ate.OTG().PushConfig(t, topo)
+}
+
+func startCapture(t *testing.T, ate *ondatra.ATEDevice) gosnappi.ControlState {
+	t.Helper()
+	cs := gosnappi.NewControlState()
+	cs.Port().Capture().SetState(gosnappi.StatePortCaptureState.START)
+	ate.OTG().SetControlState(t, cs)
+	return cs
+}
+
+func processCapture(t *testing.T, ate *ondatra.ATEDevice, capturePort string, flowName string) {
```

Comment:
https://google.github.io/styleguide/go/decisions#must-functions

Rename func to follow `must function` guidance.


```suggestion
func mustProcessCapture(t *testing.T, ate *ondatra.ATEDevice, capturePort string, flowName string) {
```